### ISSUE TITLE
Allow non-logical groups to be "visible" in hierarchy

### ISSUE THREAD

**Issue by cooperka** [getodk/collect#2818]
#### Software and hardware versions 
Collect v1.19

#### Problem description
There was a new feature added in https://github.com/opendatakit/collect/pull/2764 to allow groups to show up in the form hierarchy. Groups must be both **presentable** and **logical** (see [docs](https://opendatakit.github.io/xforms-spec/#groups) for definitions) in order to be visible.

Non-logical groups (groups without a `ref` attribute) are not currently displayed, but maybe they should be. They were originally intended to be, but the implementation didn't allow their children to be grouped properly, so we disallowed them from this new feature (at least for now).

#### Steps to reproduce the problem
1. Create a form with some grouped fields
2. Add a `label` to the group
    - The group is not yet visible in the hierarchy
3. Add a `ref` to the group
    - Now the group is visible in the hierarchy

Ideally, step 3 should not be necessary.

Example form: [visible_groups.xml](https://github.com/opendatakit/collect/files/2789364/visible_groups.xml.txt). This form has `ref`s on each group, but if you remove them, the groups should still be visible in the hierarchy view.

#### Other information 
The problem I ran into in the first place is that `TreeReference` is only generated for nodes with a `ref` (that's its purpose). For this reason, a group without a `ref` wouldn't have a unique `TreeReference` (e.g. it would just be `/data`, the same as other nodes on its level) and grouping its children becomes difficult because you can't use `TreeReference#isParentOf`.

This is probably solved quite simply by using something other than `TreeReference`. I'll look into it and see what I can do; in the meantime, I'd appreciate any insight or advice to make my job easier 😁

Relevant code link: [FormHierarchyActivity check for isDisplayableGroup](https://github.com/opendatakit/collect/blob/2182652/collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java#L614)

Tip: A solution may be able to use something similar to the logic in `FormController#indexContainsRepeatableGroup()`.

#### Alternative solution
According to https://github.com/opendatakit/build/issues/195:
> If a `group` has a corresponding node in the `instance`, it should have a `ref` attribute: https://opendatakit.github.io/xforms-spec/#groups

If **that** issue is correct, it means **this** issue is no longer a problem, because `ref` technically needs to be added to visible groups anyway, and Build should have been (and will be) doing this automatically.

**Comment by lognaturel** [comment]
I *think* you have `FormIndex` objects and could use `isSubElement`? 

**EDIT**: if the group is not in the instance (which it shouldn't be if there is no `ref`), then I don't think it has a `FormIndex` so that probably won't work.

**Comment by cooperka** [comment]
@yanokwa and I decided to deprioritize this, mainly because:

1. After opendatakit/build#195 all new forms from Build will work by default
1. Old forms will behave the way they always have and won't be affected
1. opendatakit/javarosa#402 added tests for the existing logic
1. The existing behavior had all sorts of edge cases and regressions; updating it yet again is opening a can of worms that has relatively low reward

Will be removed from the v1.20 milestone; optionally also being closed 🌱

**Comment by yanokwa** [comment]
Agreed that this is high-risk, so I'm going to remove it from the milestone. But I'm going to leave it open because it would be nice to have at some point. But yeah, low priority at this point.
